pkg/api/controller.go

```diff
@@ -99,6 +99,225 @@ type Controller struct {
 	otfDiffService        *tablediff.Service
 }
 
+func (c *Controller) CreatePresignMultipartUpload(w http.ResponseWriter, r *http.Request, repository string, branch string, params apigen.CreatePresignMultipartUploadParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.WriteObjectAction,
+			Resource: permissions.ObjectArn(repository, params.Path),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "create_presign_multipart_upload", r, repository, branch, "")
```

Comment:
Need to check `parts` here

Reply:
Didn't want to require parts as we like to enable future stream use-case where you can request to create mpu without specify number of parts and later as for them.

Reply:
At least check it's non-negative?